enumerable_methods.rb

```diff
@@ -0,0 +1,132 @@
+module Enumerable
+  # start of my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+  # end of my_each
+
+  # start of my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    until i >= length
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+  # end of my_each_with_index
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    new_array = []
+    i = 0
+
+    until i >= length
+      new_array << i if yield self[i]
+      i += 1
+    end
+  end
+  # end of my_select
+
+  # my_all?
+  def my_all?
+    my_each do |item|
+      return false unless yield item
+    end
+    true
+  end
+  # end of my_all?
+
+  # my_any?
+  def my_any?
+    my_each do |item|
+      return true if yield item
+    end
+    false
+  end
+  # end of my_any?
+
+  # my_none?
+  def my_none?
+    my_each do |item|
+      return false if yield item
+    end
+    true
+  end
+  # end of my_none?
+
+  # my_count
+  def my_count
```

Comment:
- [ ] [`count`](https://rubyapi.org/2.7/o/array#method-i-count) works by expected a block or an argument. In case of passing argument it will count the occurences of that argument.
```ruby
[1, 2, 3].count(2) # => 1
```

In case of passing a block it will return number of items yield true inside the block.

```ruby
[1, 2, 3].count(&:even?)
# or
[1, 2, 3].count { |x| x.even? }
# should give 1
```

You have implemented the block (not working right though -- `count` is increasing no matter what), but we have to implement the other case when just an argument is passed (`[1, 2, 3].my_count(2)`)